### ISSUE TITLE
waf.py does not work with python3.5

### ISSUE THREAD

**Issue by prololo** [prologin/stechec2#28]
> ###### Originally posted 2015-11-09 10:18 by [choclate](https://bitbucket.org/choclate/) on Bitbucket

Hello got this error with python3.5:

```
[choclate@choclate stechec2]$ ./waf.py configure --with-games=tictactoe --prefix=/usr
Setting top to                           : /home/choclate/documents/projects/python/stechec2 
Setting out to                           : /home/choclate/documents/projects/python/stechec2/build 
Checking for 'g++' (c++ compiler)        : /usr/lib/hardening-wrapper/bin/g++ 
Unpacking gtest                          : yes 
Traceback (most recent call last):
  File "/home/choclate/documents/projects/python/stechec2/.waf3-1.6.11-aa92258099202450c3af4004c606f61b/waflib/Scripting.py", line 97, in waf_entry_point
    run_commands()
  File "/home/choclate/documents/projects/python/stechec2/.waf3-1.6.11-aa92258099202450c3af4004c606f61b/waflib/Scripting.py", line 151, in run_commands
    run_command(cmd_name)
  File "/home/choclate/documents/projects/python/stechec2/.waf3-1.6.11-aa92258099202450c3af4004c606f61b/waflib/Scripting.py", line 143, in run_command
    ctx.execute()
  File "/home/choclate/documents/projects/python/stechec2/.waf3-1.6.11-aa92258099202450c3af4004c606f61b/waflib/Configure.py", line 128, in execute
    super(ConfigurationContext,self).execute()
  File "/home/choclate/documents/projects/python/stechec2/.waf3-1.6.11-aa92258099202450c3af4004c606f61b/waflib/Context.py", line 87, in execute
    self.recurse([os.path.dirname(g_module.root_path)])
  File "/home/choclate/documents/projects/python/stechec2/.waf3-1.6.11-aa92258099202450c3af4004c606f61b/waflib/Context.py", line 128, in recurse
    user_function(self)
  File "/home/choclate/documents/projects/python/stechec2/wscript", line 36, in configure
    conf.load('unittest_gtest')
  File "/home/choclate/documents/projects/python/stechec2/.waf3-1.6.11-aa92258099202450c3af4004c606f61b/waflib/Configure.py", line 190, in load
    if type(func)is type(Utils.readf):func(self)
  File "./tools/waf/unittest_gtest.py", line 107, in configure
    conf.check_cxx(lib = 'pthread', uselib_store = 'GTEST_PTHREAD')
  File "/home/choclate/documents/projects/python/stechec2/.waf3-1.6.11-aa92258099202450c3af4004c606f61b/waflib/Configure.py", line 221, in fun
    return f(*k,**kw)
  File "/home/choclate/documents/projects/python/stechec2/.waf3-1.6.11-aa92258099202450c3af4004c606f61b/waflib/Tools/c_config.py", line 458, in check_cxx
    return self.check(*k,**kw)
  File "/home/choclate/documents/projects/python/stechec2/.waf3-1.6.11-aa92258099202450c3af4004c606f61b/waflib/Configure.py", line 221, in fun
    return f(*k,**kw)
  File "/home/choclate/documents/projects/python/stechec2/.waf3-1.6.11-aa92258099202450c3af4004c606f61b/waflib/Tools/c_config.py", line 366, in check
    ret=self.run_c_code(*k,**kw)
  File "/home/choclate/documents/projects/python/stechec2/.waf3-1.6.11-aa92258099202450c3af4004c606f61b/waflib/Configure.py", line 221, in fun
    return f(*k,**kw)
  File "/home/choclate/documents/projects/python/stechec2/.waf3-1.6.11-aa92258099202450c3af4004c606f61b/waflib/Tools/c_config.py", line 445, in run_c_code
    bld.compile()
  File "/home/choclate/documents/projects/python/stechec2/.waf3-1.6.11-aa92258099202450c3af4004c606f61b/waflib/Build.py", line 188, in compile
    self.store()
  File "/home/choclate/documents/projects/python/stechec2/.waf3-1.6.11-aa92258099202450c3af4004c606f61b/waflib/Utils.py", line 310, in f
    ret=fun(*k,**kw)
  File "/home/choclate/documents/projects/python/stechec2/.waf3-1.6.11-aa92258099202450c3af4004c606f61b/waflib/Build.py", line 162, in store
    cPickle.dump(data,f)
AttributeError: Can't pickle local object 'Context.__init__.<locals>.node_class'
Checking for library pthread             : 
```

It works with `python3.4 waf.py configure --with-games=tictactoe --prefix=/us`.

**Comment by seirl** [comment]
> ###### Originally posted 2015-11-09 10:20 by [serialk](https://bitbucket.org/serialk/) on Bitbucket

I encountered the exact same issue yesterday. The PKGBUILD file is broken. I need to look into that.

**Comment by emersion** [comment]
> ###### Originally posted 2015-11-09 18:59 by [emersion](https://bitbucket.org/emersion/) on Bitbucket

Same problem here

**Comment by seirl** [comment]
> ###### Originally posted 2015-11-09 20:29 by [serialk](https://bitbucket.org/serialk/) on Bitbucket

You can use python2 or python3.4 as a temporary workaround.

**Comment by seirl** [comment]
> ###### Originally posted 2015-11-22 06:40 by [serialk](https://bitbucket.org/serialk/) on Bitbucket

Fixed by upgrading to waf 1.8
